java/src/main/native/src/row_conversion.cu

```diff
@@ -1263,19 +1345,29 @@ static size_type compute_column_information(iterator begin, iterator end,
     // of data here for offset/length
     auto col_size = nested_type ? 8 : size_of(col_type);
 
-    // align size for this type
-    size_type const alignment_needed = col_size; // They are the same for fixed width types
+    // align size for this type - They are the same for fixed width types and 4 bytes for variable
+    // width length/offset combos
+    size_type const alignment_needed = nested_type ? 4 : col_size;
     fixed_width_size_per_row = util::round_up_unsafe(fixed_width_size_per_row, alignment_needed);
-    column_starts.push_back(fixed_width_size_per_row);
-    column_sizes.push_back(col_size);
+    if (nested_type) {
```

Comment:
I wonder if this variable should be named `compound`.

The following is not a suggestion to change code:

This makes me wonder about the distinctions in CUDF between nested, fixed-width, and compound types.
Could a `STRUCT` column containing only fixed width members be viewed as a fixed-width type? We might conceivably avoid the cost of variable width serialization for such structs.

A question to consider at another time, maybe.

Reply:
I like the `compound` name and I've done that. I think the rest of this is more philosophical and should be considered at a later time.